web/src/containers/ConnectedWaypoint.js

```diff
@@ -0,0 +1,42 @@
+import PropTypes from 'prop-types';
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
+import { Waypoint } from 'react-waypoint';
+
+import { scrollTo } from '../actions/navigation';
+
+class ConnectedWaypoint extends Component {
+  hitWaypoint = id => () => {
+    const { scrollTo: action } = this.props;
+    action(id);
+  };
+
+  render() {
+    const { children, id } = this.props;
+    return (
+      <Fragment>
+        <Waypoint onEnter={this.hitWaypoint(id)} bottomOffset="90%" />
+        {children}
```

Comment:
The `ConnectedWaypoint` will work identically whether it has children or not.  I made it work both ways to accommodate the case where we were returning a `<Section>` and needed a waypoint there - this way, we can wrap the `<Section>` with the `<ConnectedWaypoint>` and still just return one component - no need to introduce a `<Fragment>`.  But in places where we're already returning multiple things, we don't have to wrap anything.

Reply:
This is super smart. I love how easy this makes it to use!